pybossa/view/projects.py

```diff
@@ -3215,19 +3228,35 @@ def coowners(short_name):
             # delete ids that don't exist anymore
             delete = [value for value in old_list if value not in overlap_list]
             for _id in delete:
+                # Remove the coowner.
                 project.owners_ids.remove(_id)
             # add ids that weren't there
             add = [value for value in new_list if value not in overlap_list]
             for _id in add:
                 project.owners_ids.append(_id)
 
+            # Update coowners_dict in response.
+            coowners_dict = [{"id": user.id, "fullname": user.fullname} for user in user_repo.get_users(project.owners_ids)]
+            response['coowners_dict'] = coowners_dict
+
             # save contacts
             new_list = [int(x) for x in json.loads(request.data).get('contacts', [])]
+
+            # remove any contacts that were just removed from coowners.
+            for _id in delete:
+                if _id in new_list:
+                    new_list.remove(_id)
+
             auditlogger.log_event(project, current_user, 'update', 'project.contacts',
                 project.info.get('contacts'), new_list)
             project.info['contacts'] = new_list
 
             project_repo.save(project)
+
+            # Update contacts_dict in response.
+            contacts_dict = [{"id": user.id, "fullname": user.fullname} for user in user_repo.get_users(project.info['contacts'])]
+            response['contacts_dict'] = contacts_dict
```

Comment:
So I guess there will be additional pybossa-default-theme code change to utilize the updated 'contacts_dict'.

Actually, I'm thinking the sync of contact and coowner can be implemented purely in front-end(pybossa-default-theme)

Reply:
Indeed. Fixed via https://github.com/bloomberg/pybossa-default-theme/pull/376/commits/f20945987d0267ea9bbe4d0a699388207fc86983